libsignal-service/src/lib.rs

```diff
@@ -9,3 +9,36 @@ pub const USER_AGENT: &'static str =
     concat!(env!("CARGO_PKG_NAME"), "-rs-", env!("CARGO_PKG_VERSION"));
 
 pub struct TrustStore;
+
+/// Attachment represents an attachment received from a peer
+///
+/// Source: `textsecure/textsecure.go`
+pub struct Attachment {
+    // pub R io.Reader whatever file
```

Comment:
Hmm, that sounds like a design decision; I think we can make `Attachment` generic (`Attachement<T>`), and implement any logic against any  `T: impl Read`, or maybe even [`T: impl AsyncRead` from futures](https://docs.rs/futures/0.3.5/futures/io/trait.AsyncRead.html), since everything is async anyway.

I find it bad practice in Rust to have a type that combines a logic/IO feature (Read) and real data, so maybe there's another way.

Reply:
```rust
pub struct Attachment<T> {
    pub reader: T,
    pub mime_type: String,
}
```